x-pack/legacy/plugins/security/public/views/oidc_implicit/oidc_implicit.tsx

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import chrome from 'ui/chrome';
+
+chrome
+  .setVisible(false)
+  .setRootTemplate('<div id="reactOIDCImplicitRoot" />')
+  .setRootController('oidc_implicit', ($scope: any, $window: any) => {
+    const urlFragment: string = window.location.hash;
+
+    // HACK: an ugly hack to overcome automatic Kibana URL fragment modifications.
+    const startFragmentIndex = urlFragment.startsWith('#/') ? 2 : 1;
+    const startKibanaStateIndex = urlFragment.indexOf('?_g=');
+    const endFragmentIndex = startKibanaStateIndex >= 0 ? startKibanaStateIndex : undefined;
+
+    $window.location.href = chrome.addBasePath(
```

Comment:
__note:__ I was thinking about using `POST` here, but couldn't convince myself enough since all these args are already came through URL fragment :man_shrugging: Would be glad to hear any thoughts on this.